hw-model/src/model_fpga_subsystem.rs

```diff
@@ -180,6 +183,156 @@ impl Mci {
     }
 }
 
+#[derive(Clone)]
+pub struct XI3CWrapper {
+    pub controller: Arc<Mutex<xi3c::Controller>>,
+    pub i3c_mmio: *mut u32,
+    pub i3c_controller_mmio: *mut u32,
```

Comment:
Since these fields are pub, and not gated behind a mutable reference, it seems we can't guarantee exclusive access to these points